### ISSUE TITLE
LIVE: blood loss is applied twice per tick through a second silent door that ignores stabilization and tourniquets

### ISSUE THREAD

**Issue by daiimus** [daiimus/gelatinous#2499]
**Anomaly.** Blood loss is applied **twice per medical tick** through two independent doors. One is cadence-correct, respects the stabilization and tourniquet guards, and logs. The other applies the full rate flat, respects **no guards at all**, and is silent. Net effect: bleeding drains at ~2× the authored rate on a quiet tick and 3×+ in combat — and **a dressed wound or an applied tourniquet does not stop the bleeding.**

This is **live-firing**, with proof in the splattercast logs.

### The two doors

**Door 1 — `BleedingCondition.tick_effect` (`conditions.py:263-283`). Correct.**

```python
if self._location_stabilized(medical_state):
    return
...
if self._location_tourniqueted(medical_state):
    return
rate_per_minute = self.get_blood_loss_rate()
blood_loss = rate_per_minute * elapsed_minutes
medical_state.blood_level = max(0, medical_state.blood_level - blood_loss)
```

Guards first, then `rate × elapsed` — exactly what `CONDITION_CADENCE_SPEC.md:107` mandates: *"`rate_per_minute × elapsed_minutes`, applied fractionally."* It also emits the `BLOOD_LOSS:` splattercast line.

**Door 2 — `calculate_blood_loss_rate` → `update_vital_signs` (`core.py:654-657`, `:910-912`). Wrong.**

```python
total_loss = sum(condition.get_blood_loss_rate() for condition in self.conditions)
...
blood_loss_rate = self.calculate_blood_loss_rate()
if blood_loss_rate > 0:
    self.blood_level = max(0.0, self.blood_level - blood_loss_rate)
```

No `* elapsed_minutes`. No stabilization check. No tourniquet check. No log line. It reads the same `get_blood_loss_rate()` — the guards live in `tick_effect`, not in the rate accessor, so door 2 bypasses them entirely.

**Both fire in the same tick.** `script.py:184` runs `condition.process(...)` (door 1), then `script.py:249` runs `update_vital_signs()` (door 2). These are the only two writers that decrement `blood_level` anywhere in the codebase.

### Live proof — the silent decrement is visible in the logs

From `/usr/src/game/server/logs/channel_splattercast.log*`, Brandy Kaur, one bleeding source at 8.0/tick:

```
00:15:59  loses 8.0 blood (92.0 -> 84.0)
00:16:11  loses 8.0 blood (76.0 -> 68.0)
00:16:23  loses 8.0 blood (60.0 -> 52.0)
00:16:35  loses 8.0 blood (44.0 -> 36.0)
00:16:47  loses 8.0 blood (28.0 -> 20.0)
```

Each logged interval **ends 8.0 higher than the next one starts**: 84→76, 68→60, 52→44, 36→28. Those gaps are door 2 — one extra full-rate decrement per tick, from the writer that logs nothing. The same alternating pattern appears for a thin sleek rat, Eugene Arroyo, Justin Schneider, and Jared Noble.

### Consequences

1. **Death-by-blood-loss arrives at roughly half the intended time.** Any balance pass tuned against `BLOOD_LOSS_PER_SEVERITY` is tuning the wrong number.
2. **Stabilization and tourniquets don't work.** `constants.py:191` documents wound care as a *"full stop at the location"*; `conditions.py:266-268` documents a tourniquet as *"no loss"*. Door 2 keeps draining at the full severity rate regardless. A dressed or tourniqueted patient still bleeds out.
3. **Combat multiplies it.** `update_vital_signs` is called from at least six non-tick sites — every damaging hit (`utils.py:542`, reached from `armor_mixin.take_damage`), every substance dose (`registry.py:635`), installs and `apply_vital_consequences` (`procedures.py:1513`, `:2131-2133`), sever/decap (`items.py:2395-2397`, `:2830-2832`). **Each incoming hit charges another full minute of bleeding on top of its own damage.**
4. **It breaks the cadence spec's central guarantee** that sampling interval is not a balance lever: door 1 is interval-independent, door 2 is charged once per call, so changing `MEDICAL_TICK_INTERVAL` silently changes the effective bleed rate.
5. **Unit collision in the same sum.** Bleeding contributes a per-**minute** rate while `RenalFailureCondition.get_blood_loss_rate()` (`conditions.py:677-681`) deliberately contributes a per-**update** quantity (`RENAL_FAILURE_DECLINE_PER_TICK`, commented *"%/update blood-equiv drain"*). `calculate_blood_loss_rate` adds them together.

`constants.py:241` labels the table *"% blood loss per tick"* while its own consumer at `conditions.py:353` documents it as *"Per-minute blood loss"* — the two doors disagree in their comments as well as their code.

**Confidence and provenance:**
- **Verified by reading**: `conditions.py` end to end; `core.py:640-935`; `script.py:146-306`; `utils.py` `apply_damage_to_location`; `constants.py` in full; `CONDITION_CADENCE_SPEC.md` §3-§5. Every writer of `blood_level` enumerated repo-wide — exactly two exist. Every `update_vital_signs` call site enumerated.
- **Verified live**: the splattercast excerpts above, across five different characters. The arithmetic gap between logged intervals is the unlogged second decrement.
- **Verified by execution**: reproduced offline in a scratch harness with no DB — a severity-7 bleed drains 5.0 via `tick_effect` while simultaneously reporting 5.0 to the sum consumer.
- **Verified identical in live dir and dev clone**: md5 of `conditions.py` and `core.py` match, so this is the running code.
- **Inferred, NOT verified**: that a *stabilized or tourniqueted* character keeps bleeding. No such character exists in the DB right now, so that specific consequence is read from the code (door 2 has no gate) rather than observed. **One repro settles it**: dress a wound on a test character and watch `blood_level` across two ticks with zero `BLOOD_LOSS` log lines.
- **Design call needed**: which door is intended. The cadence spec's taxonomy endorses the `tick_effect` form, which suggests `core.py:912` is the accidental one — but I found no spec line explicitly retiring `calculate_blood_loss_rate`. `update_vital_signs`' own docstring (`core.py:668-676`) says pain/blood/consciousness *"flow through"* it, which is ambiguous between recompute and apply.

**Proposed action.** Almost certainly: make `update_vital_signs` a pure **recompute** — delete the `blood_level` decrement at `core.py:910-912` and let `tick_effect` be the sole applier, since it is the one that honours the guards and the cadence contract. Then either retire `calculate_blood_loss_rate` or reduce it to a read-only reporting helper.

Whichever way it goes, the two rate consumers must agree on units, and the `constants.py:241` "per tick" comment needs to match.

**Discovery note:** found by an automated end-to-end read of `conditions.py` and `constants.py`, independently confirmed by a second agent tasked with refuting it, then corroborated against live logs. Two of the 17 verified findings from that run turned out to be the same root cause seen from opposite ends — filed here as one issue.

Related: #2494 (`update_vital_signs`' other problem — it only *sometimes* invalidates the death cache, via this very `if blood_loss_rate > 0` branch), #2495.
